UI/js/controllers.js

```diff
@@ -130,26 +130,31 @@ dashboardApp.controller('termsController', ['Courses', 'Terms', '$rootScope', '$
     if (data.Result.length === undefined ){
       data.Result = [].concat(data.Result);
     }
-    $scope.terms = data.Result;
-    $scope.$parent.term = data.Result[0].TermDescr;
-    $scope.$parent.termId = data.Result[0].TermCode;
+    if (data.Result.length !==0)  {
 
-    $scope.courses = [];
-    $scope.loading = true;
-    var url = 'courses/' + $rootScope.user + '.json?TERMID=' + $scope.$parent.termId;
+      $scope.terms = data.Result;
+      $scope.$parent.term = data.Result[0].TermDescr;
+      $scope.$parent.termId = data.Result[0].TermCode;
 
-  //use the Courses factory as a promise. Add returned data to the scope.
+      $scope.courses = [];
+      $scope.loading = true;
+      var url = 'courses/' + $rootScope.user + '.json?TERMID=' + $scope.$parent.termId;
 
-    Courses.getCourses(url).then(function (data) {
-      if (data.failure) {
-        $scope.courses.errors = data;
-        $scope.loading = false;
-      } else {
-        $scope.courses = data;
-        $scope.loading = false;
-      }
-      $('.colHeader small').append($('<span id="done" class="sr-only">' + $scope.courses.length + ' courses </span>'));
-    });
+    //use the Courses factory as a promise. Add returned data to the scope.
+
+      Courses.getCourses(url).then(function (data) {
+        if (data.failure) {
+          $scope.courses.errors = data;
+          $scope.loading = false;
+        } else {
+          $scope.courses = data;
+          $scope.loading = false;
+        }
+        $('.colHeader small').append($('<span id="done" class="sr-only">' + $scope.courses.length + ' courses </span>'));
+      });
+    } else {
+      $scope.$parent.term  = 'You do not seem to have courses in any terms we know of.'
```

Comment:
This message sounds tentative.  Does the application really not able to determine the status?  The data should be pretty solid.  I'd suggest "You do not have courses in the current term".  Or "...in any of the recent terms", if there could be multiple.